libraries/AP_Compass/AP_Compass_Backend.cpp

```diff
@@ -100,12 +100,13 @@ void AP_Compass_Backend::correct_field(Vector3f &mag, uint8_t i)
         // per-motor correction is only valid for first compass
         _compass._per_motor.compensate(state.motor_offset);
     } else if (_compass._motor_comp_type == AP_COMPASS_MOT_COMP_THROTTLE) {
-        state.motor_offset = mot * _compass._thr;
+        state.motor_offset = mot * _compass._read_thr;
     } else if (_compass._motor_comp_type == AP_COMPASS_MOT_COMP_CURRENT) {
         AP_BattMonitor &battery = AP::battery();
+        // get fresh current data
         float current;
         if (battery.current_amps(current)) {
-            state.motor_offset = mot * current;
+            state.motor_offset = mot * current * _compass._thr_mot_factor;
```

Comment:
I'm not really sure this make sense. Were scaling the current reading from some unknown point in the past by the ratio of the throttle at two other unknown points in time. 

I guess the assumption is that the original throttle and current are from the same point in time, that is probably true for analog battery monitor, but possibly not the others.